packages/core/src/providers/netsuite.ts

```diff
@@ -0,0 +1,252 @@
+ *
+ *
+ * #### Configuration
+ * ```ts
+ * import { Auth } from "@auth/core"
+ * import Netsuite from "@auth/core/providers/netsuite"
+ *
+ * const request = new Request(origin)
+ * const response = await Auth(request, {
+ *   providers: [
+ *       NetSuite({
+ *         clientId: NETSUITE_CLIENT_ID,
+ *         clientSecret: NETSUITE_CLIENT_SECRET,
+ *         issuer: NETSUITE_ACCOUNT_ID, // EX: TSTDRV1234567 or 81555 for prod, and 1234567-SB1 for Sandbox accounts not "_" use "-".
+ *        // Returns the current user using the N/runtime module. This url can be a suitelet or RESTlet (Recommended)
+ *        // Using getCurrentUser(); So we match this schema returned from this RESTlet in the profile callback. (Required)
+ *         userinfo: "https://1234567.restlets.api.netsuite.com/app/site/hosting/restlet.nl?script=123&deploy=1",
+ *         // Optional
+ *         prompt: "login", // Required if you want to force the user to login every time.
+ *         scope: "restlets", // Optional defaults to "restlets rest_webservices". Enter the scope(s) you want to use followed by spaces.
+ *       })
+ *   ],
+ * })
+ * ```
+ *
+ * ### Resources
+ *
+ * - [NetSuite - Creating an Integration Record (OAuth 2.0)](https://docs.oracle.com/en/cloud/saas/netsuite/ns-online-help/section_157771733782.html#Related-Topics)
+ * - [NetSuite - Authorizing OAuth Requests](https://docs.github.com/en/developers/apps/building-oauth-apps/authorizing-oauth-apps)
+ * - [NetSuite - Configure OAuth Roles](https://docs.oracle.com/en/cloud/saas/netsuite/ns-online-help/section_157771510070.html#Set-Up-OAuth-2.0-Roles)
+ * - [Learn more about NetSuite OAuth 2.0](https://docs.oracle.com/en/cloud/saas/netsuite/ns-online-help/chapter_157769826287.html#OAuth-2.0)
+ * - [Source code](https://github.com/nextauthjs/next-auth/blob/main/packages/core/src/providers/netsuite.ts)
+ *
+ * ### Notes
+ *
+ * By default, Auth.js assumes that the NetSuite provider is
+ * based on the [OAuth 2](https://www.rfc-editor.org/rfc/rfc6749.html) specification.
+ *
+ * :::tip
+ * Make sure the `userinfo` matches the return type of the profile callback to ensure the user session gets read correctly.
+ * To override the defaults for your use case, check out [customizing a built-in OAuth provider](https://authjs.dev/guides/providers/custom-provider#override-default-options).
+ *
+ * :::
+ *
+ * :::info **Disclaimer**
+ * If you think you found a bug in the default configuration, you can [open an issue](https://authjs.dev/new/provider-issue).
+ *
+ * Auth.js strictly adheres to the specification and it cannot take responsibility for any deviation from
+ * the spec by the provider. You can open an issue, but if the problem is non-compliance with the spec,
+ * we might not pursue a resolution. You can ask for more help in [Discussions](https://authjs.dev/new/github-discussions).
+ *
+ * :::
+ */
+export default function NetSuite<P extends NetSuiteProfile>(

```

Comment:
One other tiny tweak, if we name this `options`, you can pull out whatever you need on like `213`, and then just pass the rest of the `options` in under hte `options` key on like `250` and don't need to pull out `clientId` and `clientSecret` additionally, for example. 

i.e.


```suggestion
export default function NetSuite<P extends NetSuiteProfile>(
  config: OAuthNetSuiteOptions<P>
): OAuthConfig<P> {
  const { issuer, userinfo: userInfo, prompt = 'none', scope = 'restlets rest_webservices', clientId, clientSecret } = options

  return {
    id: "netsuite",
    name: "NetSuite",
    type: "oauth",
    checks: ["state"],
    authorization: {
      url: `https://${issuer}.app.netsuite.com/app/login/oauth2/authorize.nl`,
      params: {
        client_id: clientId,
        prompt: prompt,
        response_type: 'code',
        scope,
      }
    },
    token: {
      url: `https://${issuer}.suitetalk.api.netsuite.com/services/rest/auth/oauth2/v1/token`,
      params: {
        grant_type: "authorization_code",
      }
    },
    userinfo: userInfo,
    profile(profile) {
      // This is the default runtime.getCurrentUser() object returned from the RESTlet or SUITELet
      return {
        id: profile.id,
        name: profile.name,
        email: profile.email,
        location: profile.location,
        role: profile.role,
        contact: profile?.contact
      }
    },
    style: { logo: "/netsuite.png", bg: "#3a4f5f", text: "#fff" },
    options
  }
}
```

Reply:
That is fine. You can commit this suggestion if you'd like. 👍🏼  @ndom91

Reply:
@HeavenlyEntity I pushed that fix and cleaned up a few other small thigns :blush:

Can you give it another test to make sure it still works? If it does we can merge it finally haha :tada:

Reply:
@ndom91 Thank you for getting this in! I noticed the color is not right lol. Can you update that to be this color for the background _#3a4f5f_ ➡️ _**#18191f**_ ? Hope that can be committed directly. Thank you @ndom91! Wanna get it perfect before the release! 🚀